### PR TITLE
dev row constructor

### PR DESCRIPTION
**Pull request by siddontang**
base support row constructor like `select row(1, 1) > row(1,1)`

row constructor now only supports in comparison operation.

TODO in other PRs:
+ add more checks in other expressions for Row constructor. 
+ check row constructor in select list, group by, order by, having, e,g `select row(1, 1)` is not allowed. 
+ support row constructor with sub select.

### CODE REVIEW

File: parser/scanner.l [hunk 1]
```diff
@@ -557,6 +558,7 @@ sys_var		"@@"(({global}".")|({session}".")|{local}".")?{ident}
 {right}			return right
 {rollback}		lval.item = string(l.val)
 			return rollback
+{row}			return row
```
Comment (shenli) [inline parser/scanner.l]: row is an unreserved keyword. Please treat is as other unreserved keyword, like "rollback".

**Comment by siddontang** [comment]
PTAL

File: expression/expressions/helper.go [hunk 2]
```diff
@@ -467,3 +475,45 @@ func EvalBoolExpr(ctx context.Context, expr expression.Expression, m map[interfa
 
 	return x != 0, nil
 }
+
+// CheckOneColumn checks whether expression e has only one column for the evaluation result.
+// Now most of the expressions have one column except Row expression.
+func CheckOneColumn(e expression.Expression) error {
+	_, ok := e.(*Row)
```
Comment (shenli) [inline expression/expressions/helper.go]: Not need to check if len(e.values) == 1?
Or why not rename this function to "IsRowExpr" and return bool?

Comment (siddontang) [inline expression/expressions/helper.go]: no need, row(1) is valid. 

because later we may check subselect expression, not only row.

Comment (siddontang) [inline expression/expressions/helper.go]: row(1) will be parsed failed in parser

File: parser/parser.y [hunk 3]
```diff
@@ -1699,6 +1700,17 @@ Operand:
 		l.ParamList = append(l.ParamList, pm)
 		$$ = pm
 	}
+|	"ROW" '(' Expression ',' ExpressionList ')'
```
Comment (shenli) [inline parser/parser.y]: Add a test cast in parser_test.go.
Is this valid syntax "row (expr)" which with no ',' in parenthesis?

File: expression/expressions/helper.go (hunk 2 above)
Comment (shenli) [inline expression/expressions/helper.go]: ok

**Comment by siddontang** [comment]
PTAL @shenli

File: parser/parser.y (hunk 3 above)
Comment (siddontang) [inline parser/parser.y]: row(expr) is invalid

**Comment by shenli** [comment]
LGTM

File: expression/expressions/isnull.go [hunk 4]
```diff
@@ -58,10 +59,14 @@ func (is *IsNull) String() string {
...
+		return nil, errors.Trace(err)
+	}
+
+	val, err := is.Expr.Eval(ctx, args)
 	if err != nil {
 		return
 	}
 
-	return Val == nil != is.Not, nil
+	return val == nil != is.Not, nil
```
Comment (qiuyesuifeng) [inline expression/expressions/isnull.go]: should this line split into two lines？

Comment (siddontang) [inline expression/expressions/isnull.go]: maybe we just need a parenthesis

**Comment by coocood** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**